### DISCUSSION TITLE
Vertices from raw arrays

### DISCUSSION THREAD

**Discussion by bsomps** [gempy-project/gempy#899] [Q&A] [answered]
Hello! I am running into a challenge regarding the vertex data output from data.solutions.raw_arrays.vertices. From what I can tell, it appears that the vertex data (x, y, z arrays) is scaled between -1 and 1, but I am struggling to accurately convert these scaled values back to their original spatial coordinates or find how they were transformed in the first place. What would be the correct approach or formula to convert these scaled coordinates back to their original spatial dimensions? Sorry if this is outlined clearly in the documentation, im having trouble finding it

**Comment by javoha** [comment]
Hi,
good question and definitely not well documented. So the easiest way to get the original vertices is not going via `data.solutions.raw_arrays`, but instead uing the "input" in `data.grid.regular_grid.values`. This contains the original coordinates of the grid. I think this is what you are looking for right?
The transformation information is in `data.transform`, but I would need to check the exact application.

**Comment by bsomps** [comment]
Thanks for your response! Im trying to access the vertices and edges of the interpolated surfaces, for reconstruction in blender which I can do via:

```python
# Access the vertices and edges from the solutions object
all_vertices = data.solutions.raw_arrays.vertices
all_edges = data.solutions.raw_arrays.edges

# access Blender's context
scene = bpy.context.scene

for i, vertices in enumerate(all_vertices):
    # Create a new mesh and object for each set of vertices
    mesh = bpy.data.meshes.new(f'GeoModel_{i}')
    obj = bpy.data.objects.new(f'GeoModel_{i}', mesh)
```

Where it creates a mesh for each set of vertices which is nice because then I have a separate object for each surface. But the only issue is that it is scaled down to values between -1 and 1 (I think) so very small and not in the right spatial location. The `data.grid.regular_grid.values` gives me the correct location for vertices for the grid, but maybe not a way to separate out the surfaces with connectivity information like edges or faces?

**Reply by javoha** [reply, marked as answer]
Ah ok for the meshes it is a different story, sorry got that wrong. I am pretty sure that the `.transofrm` works here:

```
data.transform.apply_inverse(data.solutions.raw_arrays.vertices[2])
```

The index refers to you specific mesh in teh structural frame.

**Reply by bsomps** [reply]
Yes! that's it thanks
